aws/resource_aws_codedeploy_deployment_group.go

```diff
@@ -630,6 +896,85 @@ func alarmConfigToMap(config *codedeploy.AlarmConfiguration) []map[string]interf
 	return result
 }
 
+// deploymentStyleToMap converts a codedeploy.DeploymentStyle object
+// into a []map[string]interface{} list containing a single item
+func deploymentStyleToMap(style *codedeploy.DeploymentStyle) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, 1)
+
+	if style == nil {
+		return result
+	}
+
+	item := make(map[string]interface{})
+	item["deployment_option"] = *style.DeploymentOption
+	item["deployment_type"] = *style.DeploymentType
+	result = append(result, item)
+	return result
+}
+
+// loadBalancerInfoToMap converts a codedeploy.LoadBalancerInfo object
+// into a []map[string]interface{} list containing a single item
+func loadBalancerInfoToMap(loadBalancerInfo *codedeploy.LoadBalancerInfo) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, 1)
+
+	if loadBalancerInfo == nil {
+		return result
+	}
+
+	elbs := make([]interface{}, 0, len(loadBalancerInfo.ElbInfoList))
+	for _, elb := range loadBalancerInfo.ElbInfoList {
+		item := make(map[string]interface{})
+		item["name"] = *elb.Name
+		elbs = append(elbs, item)
+	}
+
+	targetGroups := make([]interface{}, 0, len(loadBalancerInfo.TargetGroupInfoList))
+	for _, targetGroup := range loadBalancerInfo.TargetGroupInfoList {
+		item := make(map[string]interface{})
+		item["name"] = *targetGroup.Name
+		targetGroups = append(targetGroups, item)
+	}
+
+	lbInfo := make(map[string]interface{})
+	lbInfo["elb_info"] = schema.NewSet(loadBalancerInfoHash, elbs)
+	lbInfo["target_group_info"] = schema.NewSet(loadBalancerInfoHash, targetGroups)
```

Comment:
It may look unintuitive I think we should be able to just use `[]interface{}` here, i.e. `elbs` and `targetGroups` directly. We can avoid the custom hashing function that way.